src/FakeItEasy/Expressions/ArgumentConstraints/EqualityArgumentConstraint.cs

```diff
@@ -32,7 +33,14 @@ public override string ToString()
                 return "\"{0}\"".FormatInvariant(stringValue);
             }
 
-            return this.ExpectedValue.ToString();
+            try
+            {
+                return this.ExpectedValue.ToString();
+            }
+            catch (Exception)
+            {
+                return this.ExpectedValue.GetType().Name;

```

Comment:
In fact, [it *is* covered in unit tests](https://github.com/FakeItEasy/FakeItEasy/blob/376ba52f4126ea47e2bcaa494348612326714e4a/tests/FakeItEasy.Tests/Core/FakeManagerTests.cs#L335)... But adding a spec for this wouldn't hurt.